### PR TITLE
Share image for gnmi and telemetry

### PR DESCRIPTION
**Pull request by ganglyu**
#### Why I did it
Share docker image to support gnmi container and telemetry container

##### Work item tracking
- Microsoft ADO **25423918**:

#### How I did it
Create telemetry image from gnmi docker image.
Enable gnmi container and disable telemetry container by default.

#### How to verify it
Run end to end test.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: slave.mk [hunk 1]
```diff
@@ -1447,6 +1447,11 @@ $(addprefix $(TARGET_PATH)/, $(SONIC_INSTALLERS)) : $(TARGET_PATH)/% : \
 			$(if $(shell ls files/build_templates/$($(docker:-dbg.gz=.gz)_CONTAINER_NAME).service.j2 2>/dev/null),\
 				$(eval $(docker:-dbg.gz=.gz)_GLOBAL = yes)
 			)
+
+			if [ x$($(docker:-dbg.gz=.gz)_CONTAINER_NAME) == x"gnmi" ]; then
+				j2 files/build_templates/telemetry.service.j2 > telemetry.service
```
Comment (qiluo-msft) [inline slave.mk:1452]: >telemetry [](http://example.com/codeflow?start=52&length=9)

Should we rename service name in order to align with container name?

File: rules/config [hunk 2]
```diff
@@ -125,8 +125,8 @@ SONIC_DPKG_CACHE_SOURCE ?= /var/cache/sonic/artifacts
 DEFAULT_VS_PREPARE_MEM = yes
 
 
-# INCLUDE_SYSTEM_TELEMETRY - build docker-sonic-telemetry for system telemetry support
-INCLUDE_SYSTEM_TELEMETRY = y
+# INCLUDE_SYSTEM_GNMI - build docker-sonic-gnmi for system gnmi support
+INCLUDE_SYSTEM_GNMI = y
```
Comment (qiluo-msft) [inline rules/config:129]: >INCLUDE_SYSTEM_GNMI [](http://example.com/codeflow?start=0&length=19)

Will you add another `INCLUDE_SYSTEM_TELEMETRY` flag and default to `n`?

File: src/sonic-config-engine/minigraph.py [hunk 3]
```diff
@@ -2040,6 +2040,9 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     results['ACL_TABLE'] = filter_acl_table_bindings(acls, neighbors, pcs, pc_members, sub_role, current_device['type'] if current_device else None, is_storage_device, vlan_members)
     results['FEATURE'] = {
         'telemetry': {
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:2042]: >telemetry [](http://example.com/codeflow?start=9&length=9)

If build time flag `INCLUDE_SYSTEM_TELEMETRY` already disabled, we should not allow runtime feature enable/disable at all.

File: slave.mk (hunk 1 above)
Comment (ganglyu) [reply slave.mk:1452]: Now we only have one docker image for gnmi, and we have two service: gnmi and telemetry, telemetry is disabled by default.

File: rules/config (hunk 2 above)
Comment (ganglyu) [reply rules/config:129]: # INCLUDE_SYSTEM_GNMI - build docker-sonic-gnmi for system gnmi support

This flag is used to build docker image, now we don't have docker-sonic-telemetry, do you think we still need INCLUDE_SYSTEM_TELEMETRY?

File: src/sonic-config-engine/minigraph.py (hunk 3 above)
Comment (ganglyu) [reply src/sonic-config-engine/minigraph.py:2042]: INCLUDE_SYSTEM_TELEMETRY is used to build docker image for telemetry, but now we don't have docker-sonic-telemetry.

File: files/build_templates/telemetry.service.j2 [hunk 4]
```diff
@@ -10,7 +10,7 @@ StartLimitBurst=3
 
 [Service]
 User={{ sonicadmin_user }}
-ExecStartPre=/usr/local/bin/{{docker_container_name}}.sh start
-ExecStart=/usr/local/bin/{{docker_container_name}}.sh wait
-ExecStop=/usr/local/bin/{{docker_container_name}}.sh stop
```
Comment (qiluo-msft) [inline files/build_templates/telemetry.service.j2:15]: >docker_container_name [](http://example.com/codeflow?start=26&length=21)

Is it possible to keep the template in this j2 file, and render it for either telemetry container or gnmi container? You may need to use template var for "Description".

File: slave.mk (hunk 1 above)
Comment (qiluo-msft) [reply slave.mk:1452]: The current reality is one docker image, two docker containers, two services. The build infra has some difficulty to handle this mapping.

On proposal: use two docker image, two docker containers, two services. You may construct the telemetry image FROM gnmi image, with nothing extra added. Then the current build infra will easily support it.

Comment (ganglyu) [reply slave.mk:1452]: Great suggestion!

File: src/sonic-config-engine/minigraph.py [hunk 5]
```diff
@@ -2040,6 +2040,9 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
...
+            'state': 'disabled'
+        },
+        'gnmi': {
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:2045]: >gnmi [](http://example.com/codeflow?start=9&length=4)

We will not modify minigraph.py to enable new features anymore. @wen587 to review.

Comment (ganglyu) [reply src/sonic-config-engine/minigraph.py]: I have removed related changes.

**Review by qiluo-msft — APPROVED** [review]

**Review by wen587 — APPROVED** [review]
no change in minigraph.py

File: dockers/docker-sonic-gnmi/telemetry_vars.j2 [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (zbud-msft) [inline dockers/docker-sonic-gnmi/telemetry_vars.j2:1]: Have we already added GNMI table to config_db?

**Review by zbud-msft — APPROVED** [review]

**Pull request merged**